### PR TITLE
(hold) Adding Static analysis to our CI/CD pipelines

### PR DESCRIPTION
**Pull request by stevengubler**
Changes:
* Clippy (the static analysis tool) is added as a component on the codebuild instance to add `cargo clippy` support
* The test and test-dry targets in `devops/Makefile` now run static analysis with clippy.
* Fixed some low hanging fruit that clippy caught

Currently, clippy will only output warnings. There are a couple of major issues that it will throw errors for by default, but those are disabled currently, because I'm not the best guy to fix them, and we don't want to break the pipeline.

Once these issues are fixed, we can remove the block on the errors (see the top of `devops/Makefile`).

### CODE REVIEW

**Comment by cam-parra** [comment]
@felippeb Please review. If this is meant to not pass the CI then let's have a vote to force merge

File: libsovtoken/src/api/mod.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ pub extern "C" fn create_payment_address_handler(
     let handler = CreatePaymentHandler::new(CryptoSdk {});
     let ec = handler.create_payment_address_async(wallet_handle, config, payment_closure);
     trace!("api::create_payment_address_handler << result: {:?}", ec);
-    return ec as i32;
+    ec as i32
 }
```
Comment (mattraffel) [inline libsovtoken/src/api/mod.rs]: we want to leave these returns for now.   the team that started libsovtoken made the decision to always use return, even though rust allows the syntax to not use it

**Review by mattraffel — CHANGES_REQUESTED** [review]
The changes in the rust files are undoing style/format decisions the team made earlier.   Until we have committed to a new style format, let's leave this as is.

**Comment by cam-parra** [comment]
Please ignore my last statement. @stevengubler this is something that all maintainers must agree on. I see value in a uniform coding style but I do not if this is exactly what the code base needs.

File: libsovtoken/src/api/mod.rs (hunk 1 above)
Comment (stevengubler) [reply libsovtoken/src/api/mod.rs]: Sounds good. I just wanted to get clippy's output to a reasonable level, but if it'll cause more problems, let's not worry about if for now.

**Comment by stevengubler** [comment]
Reverted the minor changes.

**Review by mattraffel — APPROVED** [review]

**Comment by stevengubler** [comment]
_...bump?_

**Pull request closed without merging**